bindings/azure/openai/openai.go

```diff
@@ -0,0 +1,324 @@
+}
+
+type ChatSettings struct {
+	Temperature      float32 `mapstructure:"temperature"`
+	MaxTokens        int32   `mapstructure:"maxTokens"`
+	TopP             float32 `mapstructure:"topP"`
+	N                int32   `mapstructure:"n"`
+	PresencePenalty  float32 `mapstructure:"presencePenalty"`
+	FrequencyPenalty float32 `mapstructure:"frequencyPenalty"`
+}
+
+// MessageArray type for chat completion API.
+type MessageArray struct {
+	Messages []Message
+}
+
+type Message struct {
+	Role    string
+	Message string
+}
+
+// Prompt type for completion API.
+type Prompt struct {
+	Prompt string
+}
+
+// NewOpenAI returns a new OpenAI output binding.
+func NewOpenAI(logger logger.Logger) bindings.OutputBinding {
+	return &AzOpenAI{
+		logger: logger,
+	}
+}
+
+// Init initializes the OpenAI binding.
+func (p *AzOpenAI) Init(ctx context.Context, meta bindings.Metadata) error {
+	m := openAIMetadata{}
+	err := metadata.DecodeMetadata(meta.Properties, &m)
+	if err != nil {
+		return fmt.Errorf("error decoding metadata: %w", err)
+	}
+	if m.Endpoint == "" {
+		return fmt.Errorf("required metadata not set: %s", Endpoint)
+	}
+	if m.DeploymentID == "" {
+		return fmt.Errorf("required metadata not set: %s", DeploymentID)
+	}
+
+	if m.APIKey != "" {
+		// use API key authentication
+		var keyCredential azopenai.KeyCredential
+		keyCredential, err = azopenai.NewKeyCredential(m.APIKey)
+		if err != nil {
+			return fmt.Errorf("error getting credentials object: %w", err)
+		}
+
+		p.client, err = azopenai.NewClientWithKeyCredential(m.Endpoint, keyCredential, m.DeploymentID, nil)
+		if err != nil {
+			return fmt.Errorf("error creating Azure OpenAI client: %w", err)
+		}
+	} else {
+		// fallback to Azure AD authentication
+		settings, innerErr := azauth.NewEnvironmentSettings(meta.Properties)
+		if innerErr != nil {
+			return fmt.Errorf("error creating environment settings: %w", innerErr)
+		}
+
+		token, innerErr := settings.GetTokenCredential()
+		if innerErr != nil {
+			return fmt.Errorf("error getting token credential: %w", innerErr)
+		}
+
+		p.client, err = azopenai.NewClient(m.Endpoint, token, m.DeploymentID, nil)
+		if err != nil {
+			return fmt.Errorf("error creating Azure OpenAI client: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Operations returns list of operations supported by OpenAI binding.
+func (p *AzOpenAI) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{
+		ChatCompletionOperation,
+		CompletionOperation,
+	}
+}
+
+// Invoke handles all invoke operations.
+func (p *AzOpenAI) Invoke(ctx context.Context, req *bindings.InvokeRequest) (resp *bindings.InvokeResponse, err error) {
+	if req == nil || len(req.Metadata) == 0 {
+		return nil, fmt.Errorf("invalid request: metadata is required")
+	}
+
+	startTime := time.Now().UTC()
+	resp = &bindings.InvokeResponse{
+		Metadata: map[string]string{
+			"operation": string(req.Operation),
+			"startTime": startTime.Format(time.RFC3339),
```

Comment:
I have reverted these to `start-time` and to use RFC3339Nano as you originally did, because I later realized that other components do the same